ui/lib/core/addon/components/form-field-groups.js

```diff
@@ -22,24 +22,21 @@ import layout from '../templates/components/form-field-groups';
  *  @validationMessages={{validationMessages}}
  * />
  * ```
- *
- * @param [renderGroup=null] {String} - An allow list of groups to include in the render.
- * @param model=null {DS.Model} - Model to be passed down to form-field component. If `fieldGroups` is present on the model then it will be iterated over and groups of `FormField` components will be rendered.
- * @param onChange=null {Func} - Handler that will get set on the `FormField` component.
- * @param onKeyUp=null {Func} - Handler that will set the value and trigger validation on input changes
- * @param validationMessages=null {Object} Object containing validation message for each property
+ * @callback onChangeCallback
+ * @callback onKeyUpCallback
+ * @param {Model} model- Model to be passed down to form-field component. If `fieldGroups` is present on the model then it will be iterated over and groups of `FormField` components will be rendered.
+ * @param {string} [renderGroup] - An allow list of groups to include in the render.
+ * @param {onChangeCallback} [onChange] - Handler that will get set on the `FormField` component.
+ * @param {onKeyUpCallback} [onKeyUp] - Handler that will set the value and trigger validation on input changes
+ * @param {ModelValidations} [modelValidations] - Object containing validation message for each property
  *
  */
 
-export default Component.extend({
-  layout,
```

Comment:
you were able to remove the layout but not extend it as we've done in other addon glimmerized components? How is this not breaking the component?

Reply:
Example: 
```
class StatTextComponent extends Component {}

export default setComponentTemplate(layout, StatTextComponent);
```

Reply:
Is it because you removed the hbs files from the templates directory?

Reply:
Yes if you move the template file so that it is colocated with the component js file ember figures it out! I would love to colocate all components and even use `pod` structure so that route/controller/template files exist together in the tree.

Reply:
That's awesome. Going forward I'll try and remember this. It's nice to have some examples in the code.